### PR TITLE
[SIP-5] Refactor MapBox

### PR DESCRIPTION
**Pull request by kristw**
- Break MapBox into smaller files
- Enable `renderTrigger` for viewport changing
- Pass explicit props rather than pass all that exists in `payload.data`
- Detach `setControlValue` from the vis component
- Replace `React.createElement` with regular jsx
- Remove usage of `refs` which is deprecated.

@williaster @conglei @hughhhh @betodealmeida @mistercrunch

### CODE REVIEW

File: superset/assets/src/visualizations/MapBox/MapBox.css [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.mapbox div.widget .slice_container {
```
Comment (williaster) [inline superset/assets/src/visualizations/MapBox/MapBox.css]: could you tell if the `.widget` styles still apply without dashboard v1?

File: superset/assets/src/explore/controls.jsx [hunk 2]
```diff
@@ -1802,6 +1802,7 @@ export const controls = {
   viewport_zoom: {
     type: 'TextControl',
     label: t('Zoom'),
+    renderTrigger: true,
```
Comment (williaster) [inline superset/assets/src/explore/controls.jsx]: 💯

File: superset/assets/src/visualizations/MapBox/ScatterPlotGlowOverlay.jsx [hunk 3]
```diff
@@ -198,196 +210,26 @@ class ScatterPlotGlowOverlay extends React.Component {
     const { globalOpacity } = this.props;
     const pixelRatio = window.devicePixelRatio || 1;
     return (
-      React.createElement('canvas', {
-        ref: 'overlay',
-        width: width * pixelRatio,
-        height: height * pixelRatio,
-        style: {
+      <canvas
+        ref={(c) => { this.canvas = c; }}
```
Comment (williaster) [inline superset/assets/src/visualizations/MapBox/ScatterPlotGlowOverlay.jsx]: could we bind this in the `constructor`?

File: superset/assets/src/visualizations/MapBox/MapBox.jsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+          height={height}
+          locations={Immutable.fromJS(clusters)}
+          dotRadius={pointRadius}
+          pointRadiusUnit={pointRadiusUnit}
+          rgb={rgb}
+          globalOpacity={globalOpacity}
+          compositeOperation={'screen'}
+          renderWhileDragging={renderWhileDragging}
+          aggregatorName={aggregatorName}
+          lngLatAccessor={function (location) {
```
Comment (williaster) [inline superset/assets/src/visualizations/MapBox/MapBox.jsx]: could possibly change this to a bound / arrow function

**Review by williaster — APPROVED** [review]
LGTM overall! left a couple comments.

File: superset/assets/src/visualizations/MapBox/MapBox.css (hunk 1 above)
Comment (kristw) [reply superset/assets/src/visualizations/MapBox/MapBox.css]: I could not find any element on both explore and dashboard so I delete these css.

File: superset/assets/src/visualizations/MapBox/MapBox.jsx (hunk 4 above)
Comment (kristw) [reply superset/assets/src/visualizations/MapBox/MapBox.jsx]: Updated.

**Pull request merged**